### ISSUE TITLE
Create binding when dependency name and the origin package name are different

### ISSUE THREAD

**Issue by daisy-ycguo** [apache/incubator-openwhisk-wskdeploy#437]
Below are `dependencies` samples. 
```
dependencies: 
    myCloudant:
      location: /whisk.system/cloudant
      inputs:
        dbname: myGreatDB
   myhelloworld:
      location: github.com/daisy-ycguo/wskdeploy-test/helloworld
```

Now we can define two kinds of dependencies:
- Binding to existing package, e.g. `myCloudant` in the sample.  We will create a package binding to `/whisk.system/cloudant` with the new package name as `myCloudant`.
- Dependency to an external source code in git repo, e.g. `myhelloworld` in the sample. We will deploy the package `myhelloworld` using the manifest file under folder `http://github.com/daisy-ycguo/wskdeploy-test/helloworld`.

If the dependency name is different than the origin package name, the reference to actions under this package will look weird. For example, the origin package name is `myhelloworld`, the dependency name is `dependency_name`. When I want to refer to an action under this package, I have to use `myhelloworld/helloworld`, not `dependency_name/helloworld`. Refer to below sample.
```
dependencies: 
   dependency_name:
      location: github.com/daisy-ycguo/wskdeploy-test/helloworld
sequences:
    mySequence:
      actions: myhelloworld/helloworld
```

So my question is how to handle if the dependency name is different with the origin package name. Maybe we can deploy the origin package at first and then create a binding with the dependency name. It might be a reasonable solution.

@mrutkows what's your suggestion?

**Comment by mrutkows** [comment]
@daisy-ycguo  The intent was that the  dependency's <package_name> would be treated as a local alias for the actual package name as decribed in the referenced package.  The referenced package would have its own Manifest file that would include its actual Package name (and the one that would be used by the wskdeploy tool to replace the local alias).

**Comment by mrutkows** [comment]
@daisy-ycguo this would align with our decision on how to handle 'rules' and 'sequence' name references (that is, assume their names to be package-scoped names unless otherwise specified) in https://github.com/apache/incubator-openwhisk-wskdeploy/issues/394

**Comment by daisy-ycguo** [comment]
I verified and I think package binding can help this use case.
I made following test.
1. create package `IntegrationTestHelloWorld` by `wskdeploy -p tests/src/integration/helloworld`
2. create a binding package newpkg by `wsk package bind IntegrationTestHelloWorld newpkg`, and then get the newpkg by `wsk package get --summary newpkg`, I see all the actions are listed:
```
package /guoyingc@cn.ibm.com_dev/newpkg
 action /guoyingc@cn.ibm.com_dev/newpkg/hello-world-series
 action /guoyingc@cn.ibm.com_dev/newpkg/helloSwift
 action /guoyingc@cn.ibm.com_dev/newpkg/helloPython
 action /guoyingc@cn.ibm.com_dev/newpkg/helloJava
 action /guoyingc@cn.ibm.com_dev/newpkg/helloNodejs
```
3. create another binding package newnewpkg by `wsk package bind IntegrationTestHelloWorld newnewpkg` and then get the newnewpkg by `wsk package get --summary newnewpkg`, all the actions are listed too.
4. invoke an action of newpkg by `wsk action invoke -r newpkg/helloJava`, the correct result is returned.
```
{
    "greeting": "Hello Bob!"
}
```

So I think, to this use case, we can create the original package at first, and then create binding with the new dependency name if the dependency name is different from the original package name. What's more, users can use inputs here as a binding to the original package.
